library/core/src/iter/adapters/filter_map.rs

```diff
@@ -97,7 +97,8 @@ where
             // SAFETY: Loop conditions ensure the index is in bounds.
 
             unsafe {
-                let opt_payload_at = core::intrinsics::option_payload_ptr(&val);
+                let opt_payload_at =
+                    (&val as *const Option<B>).byte_add(core::mem::offset_of!(Option<B>, Some.0));
```

Comment:
```suggestion
                    (&val as *const Option<B>).byte_add(core::mem::offset_of!(Option<B>, Some.0)).cast();
```

Probably makes sense to cast immediately, because `opt_payload_at` here is certainly not a pointer to `Option<B>`.